umbral/curvebn.py

```diff
@@ -139,19 +147,19 @@ def __pow__(self, other):
         """
         Performs a BN_mod_exp on two BIGNUMS.
 
-        WARNING: Not in constant time yet.
+        WARNING: Only in constant time if BN_FLG_CONSTTIME is set on the BN.
         """
         if type(other) == int:
             other = openssl._int_to_bn(other)
             other = CurveBN(other, None, None, None)
 
         power = openssl._get_new_BN()
         with backend._tmp_bn_ctx() as bn_ctx:
-            res = backend._lib.BN_mod_exp(
-                power, self.bignum, other.bignum, self.order, bn_ctx
-            )
-            backend.openssl_assert(res == 1)
-
+            with openssl._tmp_bn_mont_ctx(self.order) as bn_mont_ctx:
```

Comment:
I don't love the nested context managers here - we never use the outer indentation level as a control structure.  Does it maybe make sense to either make a single context manager to handle this situation or at least throw them on the same line?

Reply:
How would they be thrown into the same line?

Reply:
You can just stick a comma between them.

But I also think it probably makes sense to make a single outer context manager for this situation, no?

Reply:
Hm, perhaps. Do we ever use `_tmp_bn_mont_ctx` without the other indeed?